gitclone/checkout.go

```diff
@@ -294,7 +294,14 @@ func createCheckoutStrategy(checkoutMethod CheckoutMethod, cfg Config, patchFile
 }
 
 func selectFetchOptions(method CheckoutMethod, cloneDepth int, fetchTags, fetchSubmodules bool, filterTree bool) fetchOptions {
+	// If cloneDepth is 0, that means the user did not set a value for it,
```

Comment:
I moved this code block over from `gitclone.go` because this way it can be part of tests of `selectFetchOptions()`